src/stdlib/parse_influxdb.rs

```diff
@@ -0,0 +1,332 @@
+use std::collections::BTreeMap;
+
+use chrono::DateTime;
+use influxdb_line_protocol::{FieldValue, ParsedLine};
+
+use crate::compiler::prelude::*;
+
+fn influxdb_lines_to_metrics(line: ParsedLine) -> Vec<ObjectMap> {
+    let ParsedLine {
+        series,
+        field_set,
+        timestamp,
+    } = line;
+
+    let timestamp = timestamp.map(DateTime::from_timestamp_nanos);
+
+    let tags: Option<ObjectMap> = series.tag_set.as_ref().map(|tags| {
+        tags.iter()
+            .map(|t| (t.0.to_string().into(), t.1.to_string().into()))
+            .collect()
+    });
+
+    field_set
+        .iter()
+        .filter_map(|f| {
+            let mut metric = ObjectMap::new();
+            let measurement = &series.measurement;
+            let field_key = f.0.to_string();
+            let field_value = match f.1 {
+                FieldValue::I64(v) => v as f64,
+                FieldValue::U64(v) => v as f64,
+                FieldValue::F64(v) => v,
+                FieldValue::Boolean(v) => {
+                    if v {
+                        1.0
+                    } else {
+                        0.0
+                    }
+                }
+                // TODO: return none or return Error?
+                FieldValue::String(_) => return None,
+            };
+
+            // influxdb_line_protocol crate seems to not allow NaN float values while parsing
+            // field values and this case should not happen, but just in case, we should
+            // ignore the field.
+            let Ok(field_value) = NotNan::new(field_value) else {
+                // TODO: return none or return error?
+                return None;
+            };
+
+            let metric_name = format!("{measurement}_{field_key}");
+            metric.insert("name".into(), metric_name.into());
+
+            if let Some(tags) = tags.as_ref() {
+                metric.insert("tags".into(), tags.clone().into());
+            }
+
+            if let Some(timestamp) = timestamp {
+                metric.insert("timestamp".into(), timestamp.into());
+            }
+
+            metric.insert("kind".into(), "absolute".into());
+
+            let gauge_object: ObjectMap = [("value".into(), field_value.into())].into();
+            metric.insert("gauge".into(), gauge_object.into());
+
+            Some(metric)
+        })
+        .collect()
+}
+
+fn parse_influxdb(bytes: Value) -> Resolved {
+    let bytes = bytes.try_bytes()?;
+    let line = String::from_utf8_lossy(&bytes);
+    let parsed_line = influxdb_line_protocol::parse_lines(&line);
+
+    let metrics = parsed_line
+        .collect::<Result<Vec<_>, _>>()?
```

Comment:
I collect into a vec a lot of times (in the Vector implementation the same), in Lines 70, 79, 84. Is it okay? Or what do you recommend to avoid those?

Reply:
I think you could consider https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.try_fold here to collect until until you hit the first error, but I like that what you currently have returns an error early before trying to do mapping of the parsed lines. I think it's ok to leave as-is. We can optimize it after this is merged if needed.

Reply:
Agree!